### PR TITLE
Add imagePullSecrets

### PR DESCRIPTION
**Pull request by ctschubel**
Add the possibility to pass custom imagePullSecrets with values.yaml
Issue: #290

### CODE REVIEW

File: traefik/templates/deployment.yaml [hunk 1]
```diff
@@ -49,6 +49,10 @@ spec:
         app.kubernetes.io/managed-by: {{ .Release.Service }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      {{- if .Values.deployment.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.deployment.imagePullSecrets | indent 6 }}
```
Comment (SantoDE) [inline traefik/templates/deployment.yaml:54]: That indentation looks off?

**Review by SantoDE — CHANGES_REQUESTED** [review]
Only 1 minor nitpick :)

**Comment by Imroath** [comment]
In fact since 02.11 docker hub will introduce pull rate limiting, and everybody (especially on AKS/GKE clusters etc) will have to login to be able to pull.
This is a must have ASAP :)

File: traefik/templates/deployment.yaml (hunk 1 above)
Comment (ctschubel) [reply traefik/templates/deployment.yaml:54]: @SantoDE Thanks for the review!

The indent function of helm starts at the first character of the value so it has to be either:
```
    spec:
      {{- if .Values.deployment.imagePullSecrets }}
      imagePullSecrets:
{{ toYaml .Values.deployment.imagePullSecrets | indent 6 }}
      {{- end }}
```
or without indent function:
```
    spec:
      {{- if .Values.deployment.imagePullSecrets }}
      imagePullSecrets:
      {{ toYaml .Values.deployment.imagePullSecrets }}
      {{- end }}
```
To get this result:
```
    spec:
      imagePullSecrets:
      - name: myRegistryKeySecretName
```

**Review by SantoDE — APPROVED** [review]
LGTM 👍

**Comment by SantoDE** [comment]
Thanks @Imroath. I know, and its going to be merged soon.

**Pull request merged**